velox/expression/tests/CastExprTest.cpp

```diff
@@ -173,6 +173,37 @@ class CastExprTest : public functions::test::CastBaseTest {
     assertEqualVectors(expected, result);
   }
 
+  template <typename T>
+  void testFloatingToDecimalCasts() {
+    // floating to short decimal
+    auto input =
+        makeFlatVector<T>({-3333.03, -2222.02, -1.0, 0.00, 100, 99999.99});
+    testComplexCast(
+        "c0",
+        input,
+        makeFlatVector<int64_t>(
+            {-33'330'300, -22'220'200, -10'000, 0, 1'000'000, 999'999'900},
+            DECIMAL(10, 4)));
+
+    // floating to long decimal
+    testComplexCast(
```

Comment:
Please also add the test for the corner case and exception.
For example, test value > int64_max and <int64_max when output type is long decimal.